### PR TITLE
wrap within callback in try catch

### PR DESCRIPTION
**Pull request by imvetri**
Issue that will be fixed by the PR : https://github.com/Codeception/CodeceptJS/issues/454

### CODE REVIEW

File: lib/within.js [hunk 1]
```diff
@@ -21,18 +21,24 @@ function within(context, fn) {
       if (helpers[helper]._withinBegin) recorder.add('start within block', () => helpers[helper]._withinBegin(context));
     });
 
-    fn();
+    try{
+      fn();
+    }
+    catch(e){
+      //need suggestion how to report error with respect to codeceptJS
```
Comment (imvetri) [inline lib/within.js]: I would like to report it to user if callback fails on exception. Not sure how to do it in codecept way, hoping for guidance here.

Comment (imvetri) [inline lib/within.js]: @DavertMik travis tests are failing and I couldn't understand why. Could you please share some guidance on that ?

Comment (DavertMik) [inline lib/within.js]: Yes, looks like test failure is completely unrelated. I restarted the job

Comment (DavertMik) [inline lib/within.js]: Please use this to add error to promise chain:

```js
 recorder.throw(err); // catch error in test and rethrow into promise
```

**Comment by DavertMik** [comment]
Outdated

**Pull request closed without merging**